src/Kentico.Xperience.AiraUnified/Admin/AiraUnifiedEndpointDataSource.cs

```diff
@@ -116,12 +116,9 @@ private static Endpoint CreateAiraEndpointFromBody<T>(
     {
         var airaUnifiedController = await GetAiraUnifiedControllerInContext(context, actionName);
 
-        if (!await CheckHttps(context))
-        {
-            return;
-        }
-
-        if (requiredPermission is not null && !await AuthorizeOrSetRedirectToSignIn(context, configurationInfo.AiraUnifiedConfigurationItemAiraPathBase, requiredPermission))
+        if (!await CheckHttps(context) ||
+            (requiredPermission is not null && !await AuthorizeOrSetRedirectToSignIn(context, configurationInfo.AiraUnifiedConfigurationItemAiraPathBase, requiredPermission))
```

Comment:
One more thing - what about wrapping this condition into a separate method - it's being used on multiple places in this file thus making it a great candidate for this.